Fails to build with sdl2_ttf 2.24.0

### Platform

Linux / BSD

### Compiler and build tool versions

Clang 18.1.8

### Operating system version

Arch Linux

### Build commands used

Working custom PKGBUILD.  Specifics not relevant.  More info below.

### What happens

Problem is caused by changes in sdl2_ttf.  _TTF_Font was renamed TTF_Font.  See https://github.com/libsdl-org/SDL_ttf/commit/22347419ee08e49d77411c680f15e314ef870ab7

Changing instances in `Common/Render/Text/draw_text_sdl.h` allows build to succeed.  Have not tested functionality.

`sed -E -e 's&_TTF_Font&TTF_Font&g' -i "Common/Render/Text/draw_text_sdl.h"`

<details><summary>Errors.</summary>

```
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:190:12: error: cannot initialize a variable of type 'TTF_Font *' with an lvalue of type '_TTF_Font *'
  190 |         TTF_Font *font = fontMap_.find(fontHash_)->second;
      |                   ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.h:13:8: note: '_TTF_Font' is not defined, but forward declared here; conversion would be valid if it was derived from 'TTF_Font'
   13 | struct _TTF_Font;
      |        ^
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:215:13: error: cannot initialize a variable of type 'TTF_Font *' with an lvalue of type 'value_type' (aka '_TTF_Font *')
  215 |                 TTF_Font *fallbackFont = fallbackFonts_[i];
      |                           ^              ~~~~~~~~~~~~~~~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.h:13:8: note: '_TTF_Font' is not defined, but forward declared here; conversion would be valid if it was derived from 'TTF_Font'
   13 | struct _TTF_Font;
      |        ^
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:231:19: error: no matching member function for call to 'push_back'
  231 |                         fallbackFonts_.push_back(font);
      |                         ~~~~~~~~~~~~~~~^~~~~~~~~
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.2.1/../../../../include/c++/14.2.1/bits/stl_vector.h:1283:7: note: candidate function not viable: cannot convert argument of incomplete type 'TTF_Font *' to 'const value_type' (aka '_TTF_Font *const') for 1st argument
 1283 |       push_back(const value_type& __x)
      |       ^         ~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/14.2.1/../../../../include/c++/14.2.1/bits/stl_vector.h:1300:7: note: candidate function not viable: cannot convert argument of incomplete type 'TTF_Font *' to 'value_type' (aka '_TTF_Font *') for 1st argument
 1300 |       push_back(value_type&& __x)
      |       ^         ~~~~~~~~~~~~~~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:265:23: error: incompatible pointer types assigning to 'mapped_type' (aka '_TTF_Font *') from 'TTF_Font *'
  265 |         fontMap_[fontHash] = font;
      |                              ^~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:281:12: error: cannot initialize a variable of type 'TTF_Font *' with an lvalue of type '_TTF_Font *'
  281 |         TTF_Font *font = fontMap_.find(fontHash_)->second;
      |                   ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.h:13:8: note: '_TTF_Font' is not defined, but forward declared here; conversion would be valid if it was derived from 'TTF_Font'
   13 | struct _TTF_Font;
      |        ^
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:289:11: error: incompatible pointer types assigning to 'TTF_Font *' from 'value_type' (aka '_TTF_Font *')
  289 |                         font = fallbackFo[ 50%] Building CXX object CMakeFiles/Common.dir/Common/TimeUtil.cpp.o
nts_[fallbackFont];
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:322:12: error: cannot initialize a variable of type 'TTF_Font *' with an lvalue of type '_TTF_Font *'
  322 |         TTF_Font *font = fontMap_.find(fontHash_)->second;
      |                   ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.h:13:8: note: '_TTF_Font' is not defined, but forward declared here; conversion would be valid if it was derived from 'TTF_Font'
   13 | struct _TTF_Font;[ 50%] Building CXX object CMakeFiles/Common.dir/Common/SysError.cpp.o

      |        ^
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:330:11: error: incompatible pointer types assigning to 'TTF_Font *' from 'value_type' (aka '_TTF_Font *')
  330 |                         font = fallbackFonts_[fallbackFont];
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:386:3: error: no matching function for call to 'TTF_CloseFont'
  386 |                 TTF_CloseFont(iter.second);
      |                 ^~~~~~~~~~~~~
/usr/include/SDL2/SDL_ttf.h:2082:30: note: candidate function not viable: cannot convert argument of incomplete type '_TTF_Font *' to 'TTF_Font *' for 1st argument
 2082 | extern DECLSPEC void SDLCALL TTF_CloseFont(TTF_Font *font);
      |                              ^             ~~~~~~~~~~~~~~
/home/builder/build/src/ppsspp/Common/Render/Text/draw_text_sdl.cpp:389:3: error: no matching function for call to 'TTF_CloseFont'
  389 |                 TTF_CloseFont(iter);
      |                 ^~~~~~~~~~~~~
/usr/include/SDL2/SDL_ttf.h:2082:30: note: candidate function not viable: cannot convert argument of incomplete type '_TTF_Font *' to 'TTF_Font *' for 1st argument
 2082 | extern DECLSPEC void SDLCALL TTF_CloseFont(TTF_Font *font);
      |                              ^             ~~~~~~~~~~~~~~
10 errors generated.
```
</details>

### PPSSPP version affected

5431bb3

### Last working version

_No response_

### Checklist

- [X] Test in the [latest git build](https://www.ppsspp.org/devbuilds) in case it's already fixed.
- [X] Make sure to run `git submodule update --init --recursive` before building.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try deleting the build directory and running the build again.
- [ ] Check GitHub Actions, which builds every merge and PR.
- [ ] Include logs and help us reproduce.

--- a/Common/Render/Text/draw_text_sdl.h
+++ b/Common/Render/Text/draw_text_sdl.h
@@ -9,8 +9,7 @@
 #include <fontconfig/fontconfig.h>
 #endif
 
-// SDL2_ttf's TTF_Font is a typedef of _TTF_Font.
-struct _TTF_Font;
+struct TTF_Font;
 
 class TextDrawerSDL : public TextDrawer {
 public:
@@ -32,9 +31,9 @@ private:
 	uint32_t CheckMissingGlyph(std::string_view text);
 	int FindFallbackFonts(uint32_t missingGlyph, int ptSize);
 
-	std::map<uint32_t, _TTF_Font *> fontMap_;
+	std::map<uint32_t, TTF_Font *> fontMap_;
 
-	std::vector<_TTF_Font *> fallbackFonts_;
+	std::vector<TTF_Font *> fallbackFonts_;
 	std::vector<std::pair<std::string, int>> fallbackFontPaths_; // path and font face index
 
 	std::map<int, int> glyphFallbackFontIndex_;

--- a/Common/Render/Text/draw_text_sdl.h
+++ b/Common/Render/Text/draw_text_sdl.h
@@ -5,11 +5,14 @@
 #include <map>
 #include "Common/Render/Text/draw_text.h"
 
+#if defined(USE_SDL2_TTF)
+
+#include "SDL2/SDL.h"
+#include "SDL2/SDL_ttf.h"
+
 #if defined(USE_SDL2_TTF_FONTCONFIG)
 #include <fontconfig/fontconfig.h>
 #endif
-
-struct TTF_Font;
 
 class TextDrawerSDL : public TextDrawer {
 public:
@@ -42,3 +45,5 @@ private:
 	FcConfig *config;
 #endif
 };
+
+#endif

